### DISCUSSION TITLE
mcumgr shell command handler support

### DISCUSSION THREAD

**Discussion by bpbradley** [zephyrproject-rtos/zephyr#29407] [Ideas] [answered]
the `mcumgr-cli` tool supports shell command execution, but trying to use it on zephyr results in an unsupported error, because the actual mcumgr core doesn't implement the handlers for any rtos, for whatever reason.

I wanted this functionality so I implemented it in mcumgr, but I am not sure if my approach is right, or how I should go about getting this upstreamed.

Basically, this feature allows executing shell commands over mcumgr without physical access to the backend (in my case, the device has a USB exposed but I am not using USB CDC for the shell backend).

So I can do this now: `mcumgr -c <device> shell exec "shell command"` and it will execute the command, provide the return code, and if the dummy backend is used it will also provide the shell output.

The details are here: https://github.com/bpbradley/mynewt-mcumgr/commit/2e0165acc02920571f8b6cafe14e7803acb2cfcd

I was wondering if people here had any thoughts on it, or if this functionality should be implemented directly into zephyr instead? (perhaps as a shell backend?). Or if I should try to upstream it to the mcumgr repo, should I change the approach or is mine okay?

**Comment by de-nordic** [comment, marked as answer]
I think it is interesting and valuable addition. Can you make PR of this in apache/myenwet repo and the west manifest PR in zephyr repo so that review it?

**Reply by bpbradley** [reply]
Sure I can make a PR in that repo and go from there. I assume I should wait until the PR is merged in apache/mynewt-mcumgr before making the PR for the west manifest though right?

**Reply by de-nordic** [reply]
No, do the PR in mynewt, then do PR in zephyr that has modified west.yaml that points to the PR in myenwet. Mark it as DNM and note that it is used for testing the functionality within mcumgr, something similar to:
https://github.com/zephyrproject-rtos/zephyr/pull/26008
but with remote pointing to the mynewet repo.

**Reply by bpbradley** [reply]
Ah I see, that makes perfect sense. I will go ahead and do that then. Thanks!

**Reply by bpbradley** [reply]
Okay PRs are submitted! https://github.com/zephyrproject-rtos/zephyr/pull/29452

Thanks!
